### PR TITLE
Only dlload sysimg in one place

### LINKED ISSUE #10111: julia-debug.exe segfaults during spawn test ever since #9266
cc @vtjnash see logs, and bisect results from win32 and win64 here: https://gist.github.com/tkelman/7e6aa9bf14d382ba5b65
(skips were build failures)

### PR DESCRIPTION
**Pull request by ihnorton**
fixes #10111

### CODE REVIEW

**Comment by vtjnash** [comment]
there's too much code duplication between `jl_get_system_image_cpu_target` and `jl_restore_system_image`, since they must be exactly the same.

this if-branch should be written to only affect windows builds

**Comment by ihnorton** [comment]
Alright, refactored a little bit so there is only one call to load the sysimage. It's still hard to follow but I'm not sure how to get around the ordering requirements.

**Comment by JeffBezanson** [comment]
+1
